quartz/components/PageTitle.tsx

```diff
@@ -1,21 +1,71 @@
 import { pathToRoot } from "../util/path"
 import { QuartzComponentConstructor, QuartzComponentProps } from "./types"
 import { classNames } from "../util/lang"
+import * as fs from "fs"
+import * as path from "path"
 
-function PageTitle({ fileData, cfg, displayClass }: QuartzComponentProps) {
-  const title = cfg?.pageTitle ?? "Untitled Quartz"
-  const baseDir = pathToRoot(fileData.slug!)
-  return (
-    <h1 class={classNames(displayClass, "page-title")}>
-      <a href={baseDir}>{title}</a>
-    </h1>
-  )
+interface Options {
+  iconFolderPath: string
+  defaultIcon?: string
 }
 
-PageTitle.css = `
-.page-title {
-  margin: 0;
+const defaultOpts: Options = {
+  iconFolderPath: "",
 }
-`
 
-export default (() => PageTitle) satisfies QuartzComponentConstructor
+export default ((userOpts?: Partial<Options>) => {
+  const opts = { ...defaultOpts, ...userOpts }
+  function PageTitle({ fileData, cfg, displayClass }: QuartzComponentProps) {
+    const title = cfg?.pageTitle ?? "Untitled Quartz"
+    const iconType = fileData.frontmatter?.icon ?? opts.defaultIcon
+    const baseDir = pathToRoot(fileData.slug!)
+    if (opts.iconFolderPath.trim().length === 0 || !iconType) {
+      return (
+        <h1 class={classNames(displayClass, "page-title")}>
+          <a href={baseDir}>{title}</a>
+        </h1>
+      )
+    }
+    const iconFullPath = `${opts.iconFolderPath}/${iconType || opts.defaultIcon}.svg`
```

Comment:
hmm should these icons saved under `quartz/static`? I feel like giving too much options for users will make it very hard for ppl to know where to start

Reply:
the use case is that you want to use Obsidian to drive the content, so setting the "icon" properties with it. If the "content" is an obsidian vault (as it should be 😛 ) obsidian can't see `quart/static`